.github/workflows/build-and-push.yaml

```diff
@@ -0,0 +1,108 @@
+name: Build and Push
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-*'
+
+# env:
+  ## SET_NAME - overrides the docker image name (name is repo name with docker- prefix stripped)
+
+jobs:
+  pre:
+    runs-on: ubuntu-latest
+    outputs:
+      image: ${{ steps.set.outputs.image }}
+      kubectlVersion: ${{ steps.version.output}}
+    steps:
+      - id: set
+        run: |
+          # Computed org/repo name, docker- prefix is stripped off
+          export DEFAULT_IMAGE=$(echo -n '${{ github.event.organization.login }}/'; echo '${{ github.event.repository.name }}' | sed 's/^docker-//' | tr '[:upper:]' '[:lower:]')
+          echo "::set-output name=image::${SET_NAME:-$DEFAULT_IMAGE}"
+      - id: version
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          # Computed version (pick tag's stripped semver, i.e. only vX.Y.Z)
+          KUBECTL_VERSION=$(echo '${{ github.ref }}' | sed -E -e 's/refs\/tags\///' -e 's/(v[0-9]+.[0-9]+.[0-9]+).*/\1/')
+          echo "::set-output name=kubectlVersion::$KUBECTL_VERSION"
+
+  build:
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    needs: [pre]
+
+    strategy:
+      matrix:
+        flavor: [default, gcloud, root]
+        include:
+          - flavor: default
+            image: alpine
+          - flavor: root
+            image: alpine
+            user: root
+            suffix: -root
+          - flavor: gcloud
+            image: google/cloud-sdk:alpine
+            suffix: -gcloud
+
+    steps:
+      - uses: actions/checkout@v2
+      -
+        uses: dysnix/find-latest-tag@v1
+        id: latestTag
+        with:
+          compares-to-latest: ${{ github.ref }}
+          repository: ${{ github.repository }}
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ needs.pre.outputs.image }}
+          flavor: |
+            latest=${{ steps.latestTag.outputs.older == 'false' }}
+            suffix=${{ matrix.suffix }}
+          tags: |
+            type=semver,pattern=v{{version}}
+            type=semver,pattern=v{{major}}.{{minor}}
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.flavor }}-${{ github.sha }}
+          restore-keys: |
```

Comment:
do we need a multiline here just to easily compare ( w/ eyes, I mean ) the value with `key` value ?